java/opendcs/src/main/java/decodes/rledit/RefListFrame.java

```diff
@@ -24,1489 +37,1186 @@
+     */
+    void undoDeleteDTEButton_actionPerformed(ActionEvent e)
+    {
+        if (deletedDte != null)
+        {
+            DataType lastDT = null;
+            for(int i=0; i<deletedDte.length; i++)
+            {
+                String v = deletedDte[i];
+                if (v != null && v.length() > 0)
+                {
+                    String std = dteTableModel.getColumnName(i);
+                    DataType ndt = DataType.getDataType(std, v);
+                    if (lastDT != null)
+                        lastDT.assertEquivalence(ndt);
+                    lastDT = ndt;
+                }
+            }
+            dteTableModel.rebuild();
+            dteTableModel.fireTableDataChanged();
+        }
+        dtsChanged = true;
+        deletedDte = null;
+        undoDeleteDTEButton.setEnabled(false);
+    }
+
+    /**
+     * Deletes the selected data type equivalence and adds it to the undo buffer.
+     * Enables the undo button.
+     * @param e ActionEvent
+     */
+    void deleteDTEButton_actionPerformed(ActionEvent e)
+    {
+        int row = dteTable.getSelectedRow();
+        if (row == -1)
+        {
+            showError(labels.getString("RefListFrame.selectRowDeleteInfo"));
+            return;
+        }
+        deletedDte = (String[])dteTableModel.getRowObject(row);
+        DataTypeSet dts = Database.getDb().dataTypeSet;
+        for(int i=0; i<dteTableModel.getColumnCount(); i++)
+        {
+            String v = deletedDte[i];
+            if (v != null && v.length() > 0)
+            {
+                String std = dteTableModel.getColumnName(i);
+                DataType dt = dts.get(std, v);
+                if (dt != null)
+                    dt.deAssertEquivalence();
+            }
+        }
+        undoDeleteDTEButton.setEnabled(true);
+        dteTableModel.rebuild();
+        dteTableModel.fireTableDataChanged();
+        dtsChanged = true;
+    }
+
+    /**
+     * Called when save to DB menu item selected.
+     * @param e ActionEvent
+     */
+    void mi_saveToDb_actionPerformed(ActionEvent e)
+    {
+		final TraceDialog dlg = new TraceDialog(this, true);
+		final String CLOSE_MSG = "Saving Changes.";
+		dlg.setCloseText(CLOSE_MSG);
+		final AtomicBoolean result = new AtomicBoolean(false);
+		final Collection<DbEnum> changedEnums = EnumTab.getChanged();
+		SwingWorker<Boolean,String> worker = new SwingWorker<Boolean,String>()
 		{
-			jbInit();
-			initControls();
-		}
-		catch(Exception e) {
-			e.printStackTrace();
-		}
-
-		// Default operation is to do nothing when user hits 'X' in upper
-		// right to close the window. We will catch the closing event and
-		// do the same thing as if user had hit File - Exit.
-		setDefaultCloseOperation(WindowConstants.DO_NOTHING_ON_CLOSE);
-		addWindowListener(
-			new WindowAdapter()
+			@Override
+			protected Boolean doInBackground() throws Exception
 			{
-				public void windowClosing(WindowEvent e)
+				Database db = Database.getDb();
```

Comment:
Is the intent to remove all calls like this  Database.getDb() ?   I see several , that the github diff for this file isn't showing.

Reply:
Eventually, at this time only the EnumDAO has been setup in the new style so only the operations on the enumerations are getting changed to keep the PR smaller.